pkg/sql/executor.go

```diff
@@ -410,8 +411,21 @@ func train(wr *PipeWriter, tr *extendedSelect, db *DB, cwd string, modelDir stri
 	}
 	var program bytes.Buffer
 	if strings.HasPrefix(strings.ToUpper(tr.estimator), `XGBOOST.`) {
-		if e := genXGBoost(&program, tr, ds, fts, db); e != nil {
-			return fmt.Errorf("GenXGBoost %v", e)
+		// FIXME(weiguoz): Remove the condition after the codegen refactor
+		if os.Getenv("SQLFLOW_codegen") == "ir" {
+			ir, err := generateTrainIR(tr, db.String())
+			if err != nil {
+				return err
+			}
+			code, err := xgb.Train(ir)
```

Comment:
How about refining the interface `xgb.Train(ir)` to `xgb.Train(&program, ir)`,  that don't need to write String to the `bytes.Buffer` twice.

Reply:
I am still planning to use string because it is easier for people to understand. And we don't think too much about the performance here.

Reply:
I think either way is fine. Let's choose one and apply it for all IR codegen. :)